unit_tests/test_aggregator.py

```diff
@@ -14,11 +14,7 @@
 
 @pytest.fixture(autouse=True)
 def mock_connections(mocker):
-    """Fixture mocks methods and functions which establish external connections"""
-    mocker.patch("gsy_e_sdk.aggregator.blocking_get_request",
-                 return_value=test_agg, autospec=True)
-    mocker.patch("gsy_e_sdk.aggregator.blocking_post_request",
-                 return_value=test_agg, autospec=True)
+    """Mock methods and functions which establish external connections."""
```

Comment:
I removed the above patches because these are utils that can be used across the while module, each time with different return values, and I can clearly see that you're indeed patching these below

Reply:
+ Regarding the docstring, we have a special [guideline](https://gridsingularity.atlassian.net/wiki/spaces/D3A/pages/2406449155/Naming+and+Style+conventions#Docstrings) on how to write these, for example we decided to always use the imperative form in the docstrings

This is relevant in the rest of PR

Reply:
Regarding your first comment, I patched it for all tests because the constructor implements `_connect_to_simulation() `method, which itself executes `_create_aggregator()` and `list_aggregators()` methods. In these methods, the functions being patched you mention are called and therefore, each time I instance the `Aggregator` class, these previously patched functions would be called and therefore try to establish an external connection. 

As you noted I override them in some tests below, where I wanted to test explicitly if the particular function was called. 

I'm not sure though what you mean by the _while module_.

Reply:
Yesterday I also came up with the idea to patch private methods before patching the external functions being used inside them. However I'm not sure If this one is a good practice.

Reply:
I see, in my opinion you should create 1 method that tests the `_connect_to_simulation` (you can patch the mentioned functions here), and in the rest of the tests you can mock this entire `_connect_to_simulation` function (at this stage we already know that it's working properly)

Reply:
Well this is debatable, sometimes even private methods deserve their own testing functions.
Anyway, it's up to you, In case you wanted to refrain from testing this private method you can completely ignore the _connect_to_simulation and add an assertion in the aggregator fixture to check whether `aggregator_uuid` is defined after the initialization is complete, still you have to patch the method here

Reply:
I ended up patching these functions without an explicit return value just to avoid the execution.  Then when it was needed, I added it explicitly later on the module.